src/compas_timber/connections/half_lap.py

```diff
@@ -0,0 +1,149 @@
+        }
+        data_dict.update(Joint.data.fget(self))
+        return data_dict
+
+    @data.setter
+    def data(self, value):
+        Joint.data.fset(self, value)
+        self.beam_a_key = value["beam_a"]
+        self.beam_b_key = value["beam_b"]
+
+    @property
+    def joint_type(self):
+        return "Half-Lap"
+
+    @property
+    def beams(self):
+        return [self.beam_a, self.beam_b]
+
+
+    # TODO one function create_negative_volumes > return breps
+
+    def half_lap(beam_a, beam_b, cut):
+    
+        # Get Planes from Beams
+        plane_a = Plane.from_frame(beam_a.faces[3])
+        plane_a_side1 = Plane.from_frame(beam_a.faces[0])
+        plane_a_side2 = Plane.from_frame(beam_a.faces[2])
+        plane_b1 = Plane.from_frame(beam_b.faces[1])
+        plane_b3 = Plane.from_frame(beam_b.faces[3])
+        plane_b_side1 = Plane.from_frame(beam_b.faces[0])
+        plane_b_side2 = Plane.from_frame(beam_b.faces[2])
+
+        # Create Cut Plane
+        def cutplane(centerline_a, centerline_b, plane_a, plane_b, max_distance):
+            int_a, int_b = intersection_line_line_3D(centerline_a, centerline_b, max_distance)
+            int_a, _ = int_a
+            int_b, _ = int_b
+            point_cut = Point(*midpoint_point_point(int_a, int_b))
+            if cut == None or "A":
+                plane = plane_a
+            if cut == "B":
+                plane = plane_b
+            plane_cut = Plane(point_cut, plane[1])
+            return plane_cut
+
+        max_distance = 0.5
+        plane_cut = cutplane(beam_a.centerline, beam_b.centerline, plane_a, plane_b3, max_distance)
+
+        # Lines as Frame Intersections
+        lines = []
+        x = intersection_plane_plane(plane_a_side1, plane_b_side1)
+        lines.append(Line(x[0], x[1]))
+        x = intersection_plane_plane(plane_a_side1, plane_b_side2)
+        lines.append(Line(x[0], x[1]))
+        x = intersection_plane_plane(plane_a_side2, plane_b_side2)
+        lines.append(Line(x[0], x[1]))
+        x = intersection_plane_plane(plane_a_side2, plane_b_side1)
+        lines.append(Line(x[0], x[1]))
+
+        # Decide Cut Direction of beam_b
+        vector_angle_a_b1 = angle_vectors(plane_a[1], plane_b1[1])
+        vector_angle_a_b3 = angle_vectors(plane_a[1], plane_b3[1])
+        if vector_angle_a_b1 > vector_angle_a_b3:
+            plane_b = plane_b1
+        else:
+            plane_b = plane_b3
+
+        # Function Polyhedrons
+        def create_polyhedrons(plane1, plane2, lines):
+            points = []
+            for i in lines:
+                point = intersection_line_plane(i, plane1)
+                point = Point(*point)
+                points.append(point)
+            for i in lines:
+                point = intersection_line_plane(i, plane2)
+                point = Point(*point)
+                points.append(point)
+            return Polyhedron(points, [[0,1,2,3], [4,5,6,7], [0,1,5,4], [1,5,6,2], [2,6,7,3], [3,7,4,0]])
```

Comment:
So are these values constant (i.e. they cover all possible cases we want to handle)? If so, maybe we could try to name the faces in such way that it's clearer to someone reading the code what these numbers mean? Otherwise it just seems a bit mysterious ;)